azext_iot/tests/iothub/__init__.py

```diff
@@ -210,24 +236,335 @@ def _create_storage_account(self):
             create_account=(not settings.env.azext_iot_teststorageaccount)
         )
 
+    def _assign_storage_account_roles(self):
+        scope = "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Storage/storageAccounts/{}".format(
+            self.entity_sub,
+            EP_RG,
+            STORAGE_ACCOUNT
+        )
+        assign_rbac_role(
+            cmd=self.cmd,
+            assignee=self.entity_identity,
+            scope=scope,
+            role="Storage Blob Data Contributor",
+            max_tries=MAX_RBAC_ASSIGNMENT_TRIES
+        )
+
+        if hasattr(self, "user_identity_principal_id"):
+            assign_rbac_role(
+                cmd=self.cmd,
+                assignee=self.user_identity_principal_id,
+                scope=scope,
+                role="Storage Blob Data Contributor",
+                max_tries=MAX_RBAC_ASSIGNMENT_TRIES
+            )
+
     def _delete_storage_account(self):
         """
         Delete the storage account if it was created.
         """
         if not settings.env.azext_iot_teststorageaccount:
             self.cmd(
                 "storage account delete -n {} -g {} -y".format(
-                    self.storage_account_name, self.entity_rg
+                    STORAGE_ACCOUNT, self.entity_rg
                 ),
             )
 
         elif not settings.env.azext_iot_teststoragecontainer:
             self.cmd(
                 "storage container delete -n {} --connection-string '{}'".format(
-                    self.storage_account_name, self.storage_cstring
+                    STORAGE_ACCOUNT, self.storage_cstring
                 ),
             )
 
+    def _create_cosmos_db(self):
```

Comment:
I may have fixturized all of this so this code is prob not used.

Delete if not used